src/uct/ib/base/ib_device.c

```diff
@@ -576,7 +576,19 @@ const char *uct_ib_roce_version_str(uct_ib_roce_version_t roce_ver)
 
 const char *uct_ib_gid_str(const union ibv_gid *gid, char *str, size_t max_size)
 {
-    inet_ntop(AF_INET6, gid, str, max_size);
+    switch (uct_ib_gid_calc_len(gid)) {
+    case UCS_IPV4_ADDR_LEN:
+        inet_ntop(AF_INET, gid->raw + sizeof(*gid) - UCS_IPV4_ADDR_LEN,
+                  str, max_size);
+        break;
+    case UCS_IPV6_ADDR_LEN:
+        inet_ntop(AF_INET6, gid, str, max_size);
+        break;
+    default:
+        ucs_fatal("%s: gid %p has invalid format",
+                  ucs_status_string(UCS_ERR_INVALID_PARAM), gid);
+    }
+
```

Comment:
why needed?

Reply:
to make it human readable for any IP version

Reply:
i think we should present the GID as-is without converting it to IP, since this is the output from RDMA tools such as ibv_devinfo, show_gids, etc